[zerolib] Span<T> is readonly

Repro:
```csharp
using System;

char[] chars = new char[10];
Span<char> sc = new Span<char>(chars);
sc[0] = 'a';
```
This gives:
```console
SpanBug.cs(5,1): error CS8331: Cannot assign to property 'this' or use it as the right hand side of a ref assignment because it is a readonly variable
```
This is likely because in zero lib `Span.cs` on line 70 the index is marked as  `readony`:
```csharp
         public ref readonly T this[int index]
```
I would submit a PR, but not sure how to test that my hypothesis is correct.

--- a/src/zerolib/System/Span.cs
+++ b/src/zerolib/System/Span.cs
@@ -67,7 +67,7 @@ namespace System
             _length = length;
         }
 
-        public ref readonly T this[int index]
+        public ref T this[int index]
         {
             [Intrinsic]
             get

